src/Microsoft.OData.ModelBuilder/Conventions/Attributes/NullableAttributeExtensions.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Linq;
+
+namespace Microsoft.OData.ModelBuilder.Conventions.Attributes
+{
+    /// <summary>
+    /// Extensions for properties that have the NullableAttribute.
+    /// </summary>
+    /// <remarks>This MUST come after RequiredAttribute</remarks>
+    internal static class NullableAttributeExtensions
+    {
+        /// <summary>
+        /// Determine if the EDM PropertyConfiguration is already set to be NonNullable
+        /// </summary>
+        /// <param name="edmProperty">The <see cref="PropertyConfiguration"/> being extended.</param>
+        /// <returns><c>true</c> if NonNullable</returns>
+        internal static bool NonNullable(this PropertyConfiguration edmProperty)
```

Comment:
>edmProperty [](start = 68, length = 11)

need the null check for the input parameter?

Reply:
Fixed. As the code is written, it is already "Null safe" by using the "is" operator.

However, changed to add null checks/contracts so prevent future changes from inadvertently introducing bugs.